src/redux/explorer.js

```diff
@@ -422,26 +420,57 @@ export const explorerInitL2 = (network = null) => (dispatch, getState) => {
   if (getState().settings.network === NetworkTypes.mainnet) {
     switch (network) {
       case NetworkTypes.arbitrum:
-        // Start watching arbitrum assets
-        dispatch(arbitrumExplorerInit());
+      case NetworkTypes.polygon:
+        // Start watching arbitrum and polygon assets
+        dispatch(fetchAssetsFromRefraction());
         break;
       case NetworkTypes.optimism:
         // Start watching optimism assets
         dispatch(optimismExplorerInit());
         break;
-      case NetworkTypes.polygon:
-        // Start watching polygon assets
-        dispatch(polygonExplorerInit());
-        break;
       default:
         // Start watching all L2 assets
-        dispatch(arbitrumExplorerInit());
         dispatch(optimismExplorerInit());
-        dispatch(polygonExplorerInit());
     }
   }
 };
 
+const fetchAssetsFromRefraction = () => (dispatch, getState) => {
+  const { accountAddress, nativeCurrency } = getState().settings;
+  const { addressSocket } = getState().explorer;
+  addressSocket.emit([
+    'get',
+    {
+      payload: {
+        address: accountAddress,
+        currency: toLower(nativeCurrency),
+      },
+      scope: ['assets'],
+    },
+  ]);
+};
+
+const l2AddressAssetsReceived = (message, network) => (dispatch, getState) => {
+  const { assets: allAssets, genericAssets } = getState().data;
+
+  const updatedMessage = Object.assign({}, message);
+  updatedMessage.payload.assets = message?.payload?.assets?.map(asset => {
```

Comment:
this LHS makes me a little nervous - can we replace lines 456-469 with something like:

```
  const newAssets = message?.payload?.assets?.map(asset => {
    const mainnetAddress = toLower(asset.asset.mainnet_address);
    const fallbackAsset =
      ethereumUtils.getAsset(allAssets, mainnetAddress) ||
      genericAssets[mainnetAddress];

    if (fallbackAsset) {
      asset.asset.price = {
        ...asset.asset.price,
        ...fallbackAsset.price,
      };
    }
    return asset;
  });

  const updatedMessage = {
    ...message,
    payload: {
      ...message?.payload,
      assets: newAssets,
    } 
  };
```